### ISSUE TITLE
Syntax conflicts in search.py (tuples are used as functions!)

### ISSUE THREAD

**Issue by goswami-rahul** [aimacode/aima-python#824]
In the function `actions()` inside search.py, the variable `index_blank_square` is a `tuple` of indices, but called as a function like `index_blank_square(0)` or `index_blank_square(1)`.
```python
def actions(self, state):
        """Return the actions that can be executed in the given state.
        The result would be a list, since there are only four possible actions
        in any given state of the environment."""
       
        possible_actions = list()
        index_blank_square = self.find_blank_square(state)

        if index_blank_square(0) == 0:
            possible_actions += ['DOWN']
        elif index_blank_square(0) == 1:
            possible_actions += ['UP', 'DOWN']
        elif index_blank_square(0) == 2:
            possible_actions += ['UP']
        
        if index_blank_square(1) == 0:
            possible_actions += ['RIGHT']
        elif index_blank_square(1) == 1:
            possible_actions += ['LEFT', 'RIGHT']
        elif index_blank_square(1) == 2:
            possible_actions += ['LEFT']

        return possible_actions
```

**Comment by goswami-rahul** [comment]
I found the same issue in another function `result()`
```python
def result(self, state, action):
        """Given state and action, return a new state that is the result of the action.
        Action is assumed to be a valid action in the state."""

        blank_square = self.find_blank_square(state)
        new_state = [row[:] for row in state]

        if action=='UP':
            new_state[blank_square(0)][blank_square(1)] = new_state[blank_square(0)-1][blank_square(1)]
            new_state[blank_square(0)-1][blank_square(1)] = 0
        elif action=='LEFT':
            new_state[blank_square(0)][blank_square(1)] = new_state[blank_square(0)][blank_square(1)-1]
            new_state[blank_square(0)][blank_square(1)-1] = 0
        elif action=='DOWN':
            new_state[blank_square(0)][blank_square(1)] = new_state[blank_square(0)+1][blank_square(1)]
            new_state[blank_square(0)+1][blank_square(1)] = 0
        elif action=='RIGHT':
            new_state[blank_square(0)][blank_square(1)] = new_state[blank_square(0)][blank_square(1)+1]
            new_state[blank_square(0)][blank_square(1)+1] = 0
        else:
            print("Invalid Action!")
        return new_state
```

**Comment by ad71** [comment]
This has been fixed in #807
